### PR TITLE
[CBRD-24807] Refactor type evaluation of functions

### PR DESCRIPTION
**Pull request by hgryoo**
http://jira.cubrid.org/browse/CBRD-24807

- Rename FUNC_TYPE -> FUNC_CODE
- Move FUNC_CODE from storage_common.h to a new file
- Rename pt_eval_function_type_old () -> pt_eval_function_type_aggregate (): only aggregate functions are not migrated to the new routine yet (func_type::Node class)
- Trivial changes

### CODE REVIEW

File: src/parser/csql_grammar.y [hunk 1]
```diff
@@ -150,6 +150,8 @@ static void pt_fill_conn_info_container(PARSER_CONTEXT *parser, int buffer_pos,
 #include "memory_alloc.h"
 #include "db_elo.h"
 #include "storage_common.h"
+#include "jsp_cl.h"
```
Comment (beyondykk9) [inline src/parser/csql_grammar.y]: What does it refer by the header file? The header file is changed?

File: src/parser/func_type.cpp [hunk 2]
```diff
@@ -1470,131 +1471,1614 @@ pt_are_equivalent_types (const PT_ARG_TYPE def_type, const PT_TYPE_ENUM op_type)
...
+static int pt_character_length_for_node (PT_NODE *node, const PT_TYPE_ENUM coerce_type);
+
 /*
- * pt_get_equivalent_type () - get the type to which a node should be
- *			       converted to in order to match an expression
- *			       definition
- *   return	  : the new type
- *   def_type(in) : the type defined in the expression signature
- *   arg_type(in) : the type of the received expression argument
+ * pt_eval_function_type () -
```
Comment (beyondykk9) [inline src/parser/func_type.cpp:1486]: The comment is not matched with the real function name "pt_eval_function_type_aggregate"

Comment (beyondykk9) [inline src/parser/func_type.cpp]:
```suggestion
 * pt_eval_function_type_aggregate () -
```

File: src/parser/func_type.cpp [hunk 3]
```diff
@@ -1470,131 +1471,1614 @@ pt_are_equivalent_types (const PT_ARG_TYPE def_type, const PT_TYPE_ENUM op_type)
...
+ *   node(in): a parse tree node of type PT_FUNCTION denoting an
+ *             an expression with aggregate functions.
+ *
+ * TODO - remove me when all functions are migrated to new evaluation
+ * GIVEUP - just moved from type_check.c to type_eval_func.cpp
  */
-PT_TYPE_ENUM
-pt_get_equivalent_type (const PT_ARG_TYPE def_type, const PT_TYPE_ENUM arg_type)
+PT_NODE *
+pt_eval_function_type_aggregate (PARSER_CONTEXT *parser, PT_NODE *node)
```
Comment (beyondykk9) [inline src/parser/func_type.cpp:1496]: Where is the reference point calling this function?

File: src/query/query_aggregate.hpp [hunk 4]
```diff
@@ -29,7 +29,8 @@
 
 #include "external_sort.h"    // SORTKEY_INFO
 #include "query_list.h"
-#include "storage_common.h"   // AGGREGATE_HASH_STATE, SCAN_CODE, FUNC_TYPE
+#include "storage_common.h"   // AGGREGATE_HASH_STATE, SCAN_CODE
```
Comment (beyondykk9) [inline src/query/query_aggregate.hpp:32]: The comment looks not fit to explain the enum's.

Comment (hgryoo) [reply src/query/query_aggregate.hpp:32]: AGGREGATE_HASH_STATE and SCAN_CODE are declared in storage_common.h

File: src/parser/csql_grammar.y (hunk 1 above)
Comment (beyondykk9) [reply src/parser/csql_grammar.y]: The build works fine without jsp_cl.h.

Comment (hgryoo) [reply src/parser/csql_grammar.y]: Ok, thank you. I'll remove the header.

**Review by beyondykk9 — APPROVED** [review]

File: src/broker/shard_proxy_handler.c [hunk 5]
```diff
@@ -1001,7 +1001,7 @@ proxy_context_dump_title (FILE * fp)
   fprintf (fp,
 	   "%-5s %-10s %-10s %-5s %-5s " "%-5s %-5s %-15s " "%-15s %-15s %-15s " "%-10s %-10s %-10s " "%-10s %-10s \n",
 	   "CID", "UID", "CLIENT", "SHARD", "CAS", "BUSY", "IN_TRAN", "PREPARE_FOR_EXEC", "FREE_ON_END_TRAN",
-	   "FREE_CONTEXT", "CLIENT_END_TRAN", "FUNC_CODE", "STMT_H_ID", "STMT_HINT_TYPE", "ERROR_IND", "ERROR_CODE");
+	   "FREE_CONTEXT", "CLIENT_END_TRAN", "FUNC_TYPE", "STMT_H_ID", "STMT_HINT_TYPE", "ERROR_IND", "ERROR_CODE");
```
Comment (hyunikn) [inline src/broker/shard_proxy_handler.c]: Is this change relevant to this refactoring?

Comment (hgryoo) [reply src/broker/shard_proxy_handler.c]: Oh, It is not relevant. The function code is CAS's code. I'm going to revert it.

**Review by hyunikn — APPROVED** [review]

**Review by ctshim — APPROVED** [review]
